src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+
+import duke.task.*;
+
+import java.io.*;
+import java.nio.file.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    protected File file;
+
+    public Storage(String fileInput) throws DukeException{
+        try {
+            this.file = new File(fileInput);
+            if (!file.exists()) {
+                Files.createDirectories(Paths.get(file.getParent()));
+                this.file = Files.createFile(Path.of(fileInput)).toFile();
+            }
+        } catch (IOException err){
```

Comment:
Probably you have missed this out, but maybe you would want to have a space between the catch exception and the bracket.

```suggestion
 } catch (IOException err) {